blog/models.py

```diff
@@ -156,34 +156,39 @@ def get_header_id(self, title: str) -> int:
 
     @cached_property
     def intro(self) -> str:
-        paragraph_text = self._get_paragraphs_text_for_intro(MAX_BLOG_ARTICLE_INTRO_LENGTH)
+        paragraph_text = self._get_text_for_intro(MAX_BLOG_ARTICLE_INTRO_LENGTH)
         if len(paragraph_text) == 0:
             return "Article intro not available."
         words_cycle = cycle(paragraph_text.split())
         intro_text = self._concatenate_intro_text_from_paragraphs_text(words_cycle, MAX_BLOG_ARTICLE_INTRO_LENGTH)
         end_ellipsis = INTRO_ELLIPSIS
         return intro_text + end_ellipsis
 
-    def _get_paragraphs_text_for_intro(self, character_limit: int) -> str:
-        paragraphs: list = self._get_list_of_paragraphs()
+    def _get_text_for_intro(self, character_limit: int) -> str:
+        text_blocks: list = self._get_list_of_text_blocks()
         paragraphs_text = ""
-        if len(paragraphs) == 0:
+        if len(text_blocks) == 0:
             return paragraphs_text
 
-        paragraphs_cycle = cycle(paragraphs)
+        blocks_cycle = cycle(text_blocks)
         while len(paragraphs_text) < character_limit:
-            paragraphs_text = self._extract_paragraph_text_from_paragraph_block(paragraphs_cycle, paragraphs_text)
+            paragraphs_text = self._extract_paragraph_text_from_block(blocks_cycle, paragraphs_text)
         return paragraphs_text
 
-    def _get_list_of_paragraphs(self) -> list:
-        return list(
-            filter(lambda body_element: body_element.block.name == ArticleBodyBlockNames.PARAGRAPH.value, self.body)
-        )
+    def _get_list_of_text_blocks(self) -> list:
+        whitelisted_block_names = [ArticleBodyBlockNames.PARAGRAPH.value, ArticleBodyBlockNames.MARKDOWN.value]
+        return list(filter(lambda body_element: body_element.block.name in whitelisted_block_names, self.body))
 
     @staticmethod
-    def _extract_paragraph_text_from_paragraph_block(paragraphs: cycle, text: str) -> str:
+    def _extract_paragraph_text_from_block(blocks: cycle, text: str) -> str:
         space_between_texts = " "
-        next_text = strip_tags(next(paragraphs).value.source)
+        next_block = next(blocks)
+        if next_block.block.name == ArticleBodyBlockNames.PARAGRAPH.value:
+            source_text = next_block.value.source
+        else:
+            source_text = next_block.value
```

Comment:
Did you check what happen if user start block with something like `###` from markdown?

Reply:
I think you should check if it is mardkown block and extract text from markdown signs

Reply:
Done